### PR TITLE
TkMethods now returns 405 on unsupported method

### PR DESCRIPTION
**Pull request by victornoel**
This is for #861.

Added a test and a fix.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#875](https://github.com/yegor256/takes/issues/875) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#875](https://github.com/yegor256/takes/issues/875) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/6712fdf4-2945-49bd-89b8-e3f6cfe1ac14); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by victornoel** [comment]
@paulodamaso travis should be re-run because of #876, I don't have the access rights to do so.

**Comment by paulodamaso** [comment]
@victornoel Done

**Comment by victornoel** [comment]
@paulodamaso thanks, it worked :)

File: src/test/java/org/takes/facets/fork/TkMethodsTest.java [hunk 1]
```diff
@@ -62,4 +67,19 @@ public void throwsExceptionOnActinOnUnproperMethod() throws
...
+    /**
+     * TkMethods can return 405 status when acting on unknown method method.
+     * @throws IOException if any I/O error occurs.
+     */
+    @Test
+    public void returnsMethodIsNotAllowedForUnsupportedMethods() throws
+        IOException {
+        new FtRemote(new TkMethods(new TkEmpty(), "PUT")).exec(
+            url -> new JdkRequest(url)
+                .method("POST")
```
Comment (krzyk) [inline src/test/java/org/takes/facets/fork/TkMethodsTest.java]: @victornoel `RqMetod.POST`

File: src/test/java/org/takes/facets/fork/TkMethodsTest.java [hunk 2]
```diff
@@ -62,4 +67,19 @@ public void throwsExceptionOnActinOnUnproperMethod() throws
...
     }
+
+    /**
+     * TkMethods can return 405 status when acting on unknown method method.
+     * @throws IOException if any I/O error occurs.
+     */
+    @Test
+    public void returnsMethodIsNotAllowedForUnsupportedMethods() throws
+        IOException {
+        new FtRemote(new TkMethods(new TkEmpty(), "PUT")).exec(
```
Comment (krzyk) [inline src/test/java/org/takes/facets/fork/TkMethodsTest.java]: @victornoel `RqMethod.PUT`

File: src/test/java/org/takes/facets/fork/TkMethodsTest.java [hunk 3]
```diff
@@ -62,4 +67,19 @@ public void throwsExceptionOnActinOnUnproperMethod() throws
             new RqFake(RqMethod.GET)
         );
     }
+
+    /**
+     * TkMethods can return 405 status when acting on unknown method method.
+     * @throws IOException if any I/O error occurs.
```
Comment (krzyk) [inline src/test/java/org/takes/facets/fork/TkMethodsTest.java]: @victornoel descriptions in javadoc for throws/params/return should start with capital letter, and should not end with a dot.

**Comment by krzyk** [comment]
@victornoel thanks

**Comment by krzyk** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#875](https://github.com/yegor256/takes/issues/875) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by ypshenychka** [comment]
@krzyk According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two major issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by krzyk** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@krzyk Thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**